src/server/journal/streamer.h

```diff
@@ -106,6 +106,8 @@ class RestoreStreamer : public JournalStreamer {
   cluster::SlotSet my_slots_;
   bool fiber_cancelled_ = false;
   bool snapshot_finished_ = false;
+
+  ConditionFlag bucket_ser_;
```

Comment:
why did you add it to streamer?

Reply:
Because it has similar semantics with the `snapshot`. It traverses and serializes the buckets (see RestoreStreamer::Run)

Reply:
we dont have any preeption in serializing buckets in RestoreStreamer as we do not support yet big value serialization in migration yet